airbyte-integrations/connectors/destination-mssql/src/main/java/io/airbyte/integrations/destination/mssql/MSSQLDestination.java

```diff
@@ -0,0 +1,100 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.mssql;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.db.Databases;
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.integrations.base.Destination;
+import io.airbyte.integrations.base.IntegrationRunner;
+import io.airbyte.integrations.destination.jdbc.AbstractJdbcDestination;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MSSQLDestination extends AbstractJdbcDestination implements Destination {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MSSQLDestination.class);
+
+  public static final String DRIVER_CLASS = "com.microsoft.sqlserver.jdbc.SQLServerDriver";
+
+  public MSSQLDestination() {
+    super(DRIVER_CLASS, new MSSQLNameTransformer(), new SqlServerOperations());
+  }
+
+  @Override
+  protected JdbcDatabase getDatabase(JsonNode config) {
+    return Databases.createJdbcDatabase(
+        config.get("username").asText(),
+        config.get("password").asText(),
+        String.format("jdbc:sqlserver://%s:%s",
+            config.get("host").asText(),
+            config.get("port").asInt()),
+        "com.microsoft.sqlserver.jdbc.SQLServerDriver");
+  }
+
+  @Override
+  public JsonNode toJdbcConfig(JsonNode config) {
+    final String schema = Optional.ofNullable(config.get("schema")).map(JsonNode::asText).orElse("public");
+
+    List<String> additionalParameters = new ArrayList<>();
+
+    final StringBuilder jdbcUrl = new StringBuilder(String.format("jdbc:sqlserver://%s:%s;user=%s;password=%s;?",
+        // final StringBuilder jdbcUrl = new
+        // StringBuilder(String.format("jdbc:sqlserver://%s:%s;databaseName=%s;user=%s;password=%s;?",
+        config.get("host").asText(),
+        config.get("port").asText(),
+        // config.get("database").asText(),
+        config.get("username").asText(),
+        config.get("password").asText()));
+
+    if (config.has("ssl") && config.get("ssl").asBoolean()) {
```

Comment:
ssl on v1, nice!

Reply:
the SSL stuff is great. I think we need at least one test to check it. There are 2 options here:
1. just right a one off unit test that runs check connection with SSL turned on and sanity check that it works.
2. add a test in TestDestination that only runs if the destination is configurable with SSL. it would similarly probably just run check connection.

i don't have a really strong preference either way. 1 is probably easier 🤷‍♀️